js/ui.js

```diff
@@ -0,0 +1,228 @@
+   * @return {div}
+   */
+  this.createScheduleBox = function(schedule) {
+    // Lecture -> Classroom -> Schedule
+    var lecture = schedule.parent.parent;
+    var scheduleBoxTreeObj = {
+        tag: 'div',
+        class: ['schedule-box', 'color-' + lecture.color],
+        children: [
+        {
+          tag: 'span',
+          class: 'timespan',
+          innerHTML: schedule.timeBegin + ' ' + schedule.timeEnd
+        },
+        {
+          tag: 'span',
+          class: 'lecture-code',
+          innerHTML: lecture.code
+        }
+      ]
+    };
+
+    var scheduleBox = createHtmlElementTree(scheduleBoxTreeObj);
+
+    var timePosition = calcPositionForTime(schedule);
+    scheduleBox.style.top = timePosition.positionBegin;
+    scheduleBox.style.height = timePosition.positionEnd;
+
+    return scheduleBox;
+  }
+
+
+  /**
+   * Creates boxes that represent lecture schedules in the schedule table.
+   * <br>
+   * Generates HTML elements with this form:
+   *
+   * <pre><code>&ltdiv class="lecture-classrooms"&gt
+   *     &ltdiv class="classroom-info"&gt
+   *       &ltinput type="checkbox" id="classroom-01" name="MAC0110"&gt
+   *       &ltlabel class="classroom" for="classroom-01"&gt
+   *         &ltdiv class="classroom-number"&gtN1&lt/div&gt
+   *         &ltdiv class="classroom-teacher"&gtNome do Professor&lt/div&gt
+   *         &ltdiv class="classroom-toggle"&gt+&lt/div&gt
+   *       &lt/label&gt
+   *     &lt/div&gt
+   * &lt/div&gt</pre></code>
+   *
+   *
+   * @param {Schedule} schedule
+   * @param {String} lectureCode
+   * @return {div}
+   */
+  this.createClassroomInfo = function(classroom, lectureCode) {
+    var classroomInfoTreeObj = {
+      tag: 'div',
+      class: 'classroom-info',
+      children: [
+        {
+          tag: 'div',
+          class: 'classroom-code',
+          innerHTML: classroom.classroomCode
+        },
+        {
+          tag: 'div',
+          class: 'classroom-teacher',
+          innerHTML: classroom.teachers.join('<br>')
+        },
+        {
+          tag: 'div',
+          class: 'classroom-toggle',
+          innerHTML: 'x'
+        }
+      ]
+    };
+
+    var classroomInfo = createHtmlElementTree(classroomInfoTreeObj);
+    return classroomInfo;
+  }
+
+  /**
+   * Creates the Info section of a lecture, inside lecture explorer.
+   *
+   * @param {Lecture} lecture
+   * @return {HTML_Element} div.lecture-info
+   */
+  this.createLectureInfo = function(lecture) {
+    var lectureInfoTreeObj = {
+      tag: 'div',
+      class: ['lecture-info', 'color-' + lecture.color],
+      children: [
+        {
+          tag: 'div',
+          class: 'lecture-info-header',
+          children: [
+            {
+              tag: 'div',
+              class: 'lecture-info-header-title',
+              children: [
+                {
+                  tag: 'div',
+                  class: 'lecture-info-code',
+                  innerHTML: lecture.code
```

Comment:
Security: This probably needs to be properly sanitized or we may face a XSS attack here.